### PR TITLE
[bluetooth.am43] null annotations

### PR DESCRIPTION
**Pull request by lsiepel**
Signed-off-by: Leo Siepel <leosiepel@gmail.com>

Removed forbidden package and added null annotations. Part of multiple PR's for bluetooth bundles. Can be merged seperately.

### CODE REVIEW

File: bundles/org.openhab.binding.bluetooth.am43/src/main/java/org/openhab/binding/bluetooth/am43/internal/command/AM43Command.java [hunk 1]
```diff
@@ -121,13 +120,17 @@ public static byte getResponseHeader(byte[] response) {
...
+        byte[] value = new byte[3 + data.length];
+        value[0] = HEADER_PREFIX;
+        value[1] = header;
+        value[2] = (byte) data.length;
+        System.arraycopy(data, 0, value, 3, data.length);
+
+        byte[] combined = new byte[REQUEST_PREFIX.length + value.length + 1];
+        System.arraycopy(REQUEST_PREFIX, 0, combined, 0, REQUEST_PREFIX.length);
+        System.arraycopy(value, 0, combined, REQUEST_PREFIX.length, value.length);
+        combined[combined.length - 1] = createChecksum(value);
```
Comment (lolodomo) [inline bundles/org.openhab.binding.bluetooth.am43/src/main/java/org/openhab/binding/bluetooth/am43/internal/command/AM43Command.java:132]: You could avoid doubled table of bytes by passing a max index to `createChecksum`.

**Review by lolodomo — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.bluetooth.am43/src/main/java/org/openhab/binding/bluetooth/am43/internal/command/AM43Command.java (hunk 1 above)
Comment (lsiepel) [reply bundles/org.openhab.binding.bluetooth.am43/src/main/java/org/openhab/binding/bluetooth/am43/internal/command/AM43Command.java]: Refactored this method.

File: bundles/org.openhab.binding.bluetooth.am43/src/main/java/org/openhab/binding/bluetooth/am43/internal/command/AM43Command.java [hunk 2]
```diff
@@ -121,19 +120,20 @@ public static byte getResponseHeader(byte[] response) {
...
-        value = ArrayUtils.addAll(value, data);
-        value = ArrayUtils.add(value, createChecksum(value));
-        return ArrayUtils.addAll(REQUEST_PREFIX, value);
+        byte[] value = new byte[4 + data.length + REQUEST_PREFIX.length];
+        System.arraycopy(REQUEST_PREFIX, 0, value, 0, REQUEST_PREFIX.length);
+        value[REQUEST_PREFIX.length] = HEADER_PREFIX;
+        value[REQUEST_PREFIX.length + 1] = header;
+        value[REQUEST_PREFIX.length + 2] = (byte) data.length;
+        System.arraycopy(data, 0, value, REQUEST_PREFIX.length + 3, data.length);
+        value[value.length-1] = createChecksum(value, REQUEST_PREFIX.length, REQUEST_PREFIX.length + 2);
```
Comment (lolodomo) [inline bundles/org.openhab.binding.bluetooth.am43/src/main/java/org/openhab/binding/bluetooth/am43/internal/command/AM43Command.java:129]: The last parameter looks wrong as you are considering the content of data.

File: bundles/org.openhab.binding.bluetooth.am43/src/main/java/org/openhab/binding/bluetooth/am43/internal/command/AM43Command.java [hunk 3]
```diff
@@ -121,19 +120,20 @@ public static byte getResponseHeader(byte[] response) {
...
+        return value;
     }
 
-    protected byte createChecksum(byte[] data) {
+    protected byte createChecksum(byte[] data, int startIndex, int maxIndex) {
```
Comment (lolodomo) [inline bundles/org.openhab.binding.bluetooth.am43/src/main/java/org/openhab/binding/bluetooth/am43/internal/command/AM43Command.java:133]: `maxIndes` is probably an inappropriate name as in fact it is the first index that will not be considered.
At least it should be commented.

**Review by lolodomo — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.bluetooth.am43/src/main/java/org/openhab/binding/bluetooth/am43/internal/command/AM43Command.java (hunk 2 above)
Comment (lsiepel) [reply bundles/org.openhab.binding.bluetooth.am43/src/main/java/org/openhab/binding/bluetooth/am43/internal/command/AM43Command.java]: Good catch, thanks.

File: bundles/org.openhab.binding.bluetooth.am43/src/main/java/org/openhab/binding/bluetooth/am43/internal/command/AM43Command.java (hunk 3 above)
Comment (lsiepel) [reply bundles/org.openhab.binding.bluetooth.am43/src/main/java/org/openhab/binding/bluetooth/am43/internal/command/AM43Command.java]: i added javadoc, changed from maxindex to length. should be more clear now. thanks!

**Review by lolodomo — APPROVED** [review]
LGTM, thank you

**Pull request merged**